jaxopt/_src/osqp.py

```diff
@@ -557,7 +551,11 @@ def update(self, params, state, params_obj, params_eq, params_ineq):
                       params_precond=params_precond)
     return base.OptStep(params=sol, state=state)
 
-  def run(self, init_params, params_obj, params_eq, params_ineq) -> base.OptStep:
+  def run(self,
+          init_params: Any,
+          params_obj: Tuple[Optional[Any], Any],  # Q can be params_Q or None
```

Comment:
I don't get why it's optional. I thought the first element of the tuple is either the full matrix Q or params_Q (the parameters of matvec_Q)

Reply:
In some circumstances we could have `params_Q=None`, think to `matvec(params_Q, x)=x` when the quadratic form is just the identity matrix (quite common, for projections onto polyhedron for example).

Reply:
Ok, thanks for the clarification. We'll need to explain this clearly in the documentation.